components/ads/Advertisement.vue

```diff
@@ -151,6 +152,14 @@ export default {
         }
       }, 300)
     },
+    handleDarkModeChange() {
+      // I found issues in firefox with the store and isDark getting out of sync
+      // setTimeout seemed to fix it
+      window.setTimeout(() => {
+        this.isDark = this.$colorMode.value !== 'light'
+        this.refresh_ad()
```

Comment:
There may need to be a debounce system for ads refreshing. Too frequent ads refresh can hurt viewability rate, but it's not really important for ethicalads